### PR TITLE
VPC integration tests

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|-------------|
|      #3336       |

## Description

This PR adds tests for AWS VPC integration

#### Tier 0
- Default configuration tests
- `remove_from_bucket` parameter tests
- `only_logs_after` parameter tests
- `path` parameter tests
- `path_suffix` parameter tests
- `regions` parameter tests
- `discard_regex` parameter tests

#### Tier 1
- `only_logs_after` parameter tests
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @nico-stefani (Developer)  |           | ⚫⚫⚫ | :red_circle: :red_circle:   |         |         | [^1] [^2][^3] |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Tier 0 results
```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest --tier 0 -k vpc
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 30 items / 16 deselected / 14 selected

test_basic.py .                                                          [  7%]
test_discard_regex.py F                                                  [ 14%]
test_only_logs_after.py .F                                               [ 28%]
test_path.py F..                                                         [ 50%]
test_path_suffix.py F..                                                  [ 71%]
test_regions.py .FF                                                      [ 92%]
test_remove_from_bucket.py F                                             [100%]
=========================== short test summary info ============================
FAILED test_discard_regex.py::test_discard_regex[vpc_discard_regex] - Timeout...
FAILED test_only_logs_after.py::test_with_only_logs_after[vpc_with_only_logs_after]
FAILED test_path.py::test_path[vpc_path_with_data] - TimeoutError: The AWS mo...
FAILED test_path_suffix.py::test_path_suffix[vpc_path_suffix_with_data] - Tim...
FAILED test_regions.py::test_regions[vpc_regions_with_data] - TimeoutError: T...
FAILED test_regions.py::test_regions[vpc_inexistent_region] - TimeoutError: T...
FAILED test_remove_from_bucket.py::test_remove_from_bucket[vpc_remove_from_bucket]
====== 7 failed, 7 passed, 16 deselected, 3 warnings in 415.86s (0:06:55) ======
```
### Tier 1 results

```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest --tier 1 -k vpc
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 30 items / 29 deselected / 1 selected

test_only_logs_after.py F                                                [100%]

=========================== short test summary info ============================
FAILED test_only_logs_after.py::test_multiple_calls[vpc_only_logs_after_multiple_calls]
================ 1 failed, 29 deselected, 3 warnings in 56.08s =================

```

[^1]: Because the module is iterating over all days from `only_logs_after` to the day of execution, all *tier 0* tests are falling giving timeout. This should be fixed in his respective issue.
[^2]: Because the module freeze when is called with an inexistent region (wazuh/wazuh#15763), the test `test_regions` is failing giving a timeout. 
[^3]: Because the module is showing the message `No logs to process in bucket: ...` multiple times, the *tier 1* test is falling. This should be fixed in his respective issue.

### CODE REVIEW

File: tests/integration/test_aws/data/test_cases/remove_from_bucket_test_module/cases_remove_from_bucket.yaml [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
   configuration_parameters:
-    BUCKET_TYPE: cloudtrail
-    BUCKET_NAME: wazuh-cloudtrail-integration-tests
+    BUCKET_TYPE: vpcflow
+    BUCKET_NAME: wazuh-vpcflow-integration-tests
   metadata:
-    bucket_type: cloudtrail
-    bucket_name: wazuh-cloudtrail-integration-tests
+    bucket_type: vpcflow
+    bucket_name: wazuh-vpcflow-integration-tests
```
Comment (Selutario) [inline tests/integration/test_aws/data/test_cases/remove_from_bucket_test_module/cases_remove_from_bucket.yaml:8]: Are these changes correct? Cloudtrail will not be tested with its own bucket.

**Review by Selutario — CHANGES_REQUESTED** [review]

**Comment by Selutario** [comment]
There are some failed tests and an error:
```
======================================================================================== test session starts ========================================================================================
platform linux -- Python 3.8.10, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/selu/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: testinfra-5.0.0, html-3.1.1, metadata-2.0.4
collected 30 items / 15 deselected / 15 selected                                                                                                                                                    

test_basic.py .                                                                                                                                                                               [  6%]
test_discard_regex.py F                                                                                                                                                                       [ 13%]
test_only_logs_after.py .FFE                                                                                                                                                                  [ 33%]
test_path.py ...                                                                                                                                                                              [ 53%]
test_path_suffix.py ...                                                                                                                                                                       [ 73%]
test_regions.py Xxx                                                                                                                                                                           [ 93%]
test_remove_from_bucket.py .                                                                                                                                                                  [100%]

============================================================================================== ERRORS ===============================================================================================
```
Full output:
[results.txt](https://github.com/wazuh/wazuh-qa/files/10410492/results.txt)

File: tests/integration/test_aws/data/test_cases/remove_from_bucket_test_module/cases_remove_from_bucket.yaml (hunk 1 above)
Comment (nico-stefani) [reply tests/integration/test_aws/data/test_cases/remove_from_bucket_test_module/cases_remove_from_bucket.yaml]: Fixed in [61c7a24](https://github.com/wazuh/wazuh-qa/pull/3699/commits/61c7a24f884d3b3ec1d55f63bb7d9ff6853456f3)

File: tests/integration/test_aws/data/test_cases/discard_regex_test_module/cases_discard_regex.yaml [hunk 2]
```diff
@@ -13,3 +13,19 @@
...
+    DISCARD_FIELD: srcport
+    DISCARD_REGEX: "5319"
+  metadata:
+    bucket_type: vpcflow
+    bucket_name: wazuh-vpcflow-integration-tests
+    only_logs_after: 2022-NOV-20
+    discard_field: srcport
+    discard_regex: "5319"
+    found_logs: 5
+    skipped_logs: 1
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply tests/integration/test_aws/data/test_cases/discard_regex_test_module/cases_discard_regex.yaml:31]: Fixed in [e10b0e6](https://github.com/wazuh/wazuh-qa/pull/3699/commits/e10b0e684329ac4312902435e372e6003aec1d9a)

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 3]
```diff
@@ -94,12 +103,51 @@ def get_data_sample(self) -> dict:
...
-        }
+        })
+
+
+class VPCDataGenerator(DataGenerator):
+    BASE_PATH = f'{cons.AWS_LOGS}/{cons.RANDOM_ACCOUNT_ID}/{cons.VPC_FLOW_LOGS}/{cons.US_EAST_1_REGION}/'
+    BASE_FILE_NAME = f'{cons.RANDOM_ACCOUNT_ID}_{cons.VPC_FLOW_LOGS}_{cons.US_EAST_1_REGION}_'
+
+    def get_filename(self, prefix=None, **kwargs) -> str:
+        """Return the filename in the cloudtrail format
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py]: Fixed in [258068a](https://github.com/wazuh/wazuh-qa/pull/3699/commits/258068a66a1a5bf055be4574674df62e8841f9ac)

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 4]
```diff
@@ -94,12 +103,51 @@ def get_data_sample(self) -> dict:
                     'sharedEventID': str(uuid4())
                 }
             ]
-        }
+        })
+
+
+class VPCDataGenerator(DataGenerator):
+    BASE_PATH = f'{cons.AWS_LOGS}/{cons.RANDOM_ACCOUNT_ID}/{cons.VPC_FLOW_LOGS}/{cons.US_EAST_1_REGION}/'
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py]: Fixed in [258068a](https://github.com/wazuh/wazuh-qa/pull/3699/commits/258068a66a1a5bf055be4574674df62e8841f9ac)

**Review by Selutario — COMMENTED** [review]
The bug that appeared last time has been fixed now.

Almost all failed tests are due to timeout, which as
@nico-stefani spotted and explained, it may be because of a problem in the AWS wodle (detected in the tests) rather than a problem in the tests themselves.

However, the `test_multiple_calls` seems to be failing not because of timeout but because it receives a higher-than-expected number of logs:
```
ERROR    wazuh_testing:cli_utils.py:62 Some logs may were processed
ERROR    wazuh_testing:cli_utils.py:63 Results found: 52
ERROR    wazuh_testing:cli_utils.py:64 Results expected: 1
```

Full output: [results.txt](https://github.com/wazuh/wazuh-qa/files/10424107/results.txt)

**Review by Selutario — APPROVED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**